src/signal_scope/plotwidget.cpp

```diff
@@ -240,6 +247,47 @@ double PlotWidget::timeWindow() const
   return this->d_plot->timeWindow();
 }
 
+bool PlotWidget::getScale()
+{
+  float tmpRange[2];
+  bool update = false;
+
+  for (int ii=0; ii<mSignalListWidget->count(); ii++){
+    QListWidgetItem* signalItem = mSignalListWidget->item(ii);
+    SignalHandler* signalHandler = signalItem->data(Qt::UserRole).value<SignalHandler*>();
+    SignalData* signalData = signalHandler->signalData();
+
+    size_t nn = signalData->size();
+    if (nn){
+      tmpRange[0] = signalData->value(0).y();
+      tmpRange[1] = signalData->value(0).y();
+
+      for (size_t jj=1; jj<signalData->size(); jj++){
+        if (signalData->value(jj).y() > tmpRange[1])
+          tmpRange[1] = signalData->value(jj).y();
+        else if (signalData->value(jj).y() < tmpRange[0])
+          tmpRange[0] = signalData->value(jj).y();
+      }      
+    }
+
+    if (tmpRange[0] < yRange[0]){
+      yRange[0] = tmpRange[0];
+      update = true;
+    }
+    if (tmpRange[1] > yRange[1]){
+      yRange[1] = tmpRange[1];
+      update = true;
+    }      
+  }
+  return update;
+}
+
+void PlotWidget::resetYAxisMaxScale()
+{
+  if (getScale())
```

Comment:
maybe the getScale() function should be removed, and the update logic is encapsulated into this function resetYAxisMaxScale().  Each call to resetYAxisMaxScale() will check for an increased range and reset if the range has increased since the last call to resetYAxisMaxScale().  The getScale() function doesn't need to be in the API of PlotWidget.

Also, perhaps the cached Y range should be reset inside the call to PlotWidget::clearHistory()?